### PR TITLE
converted flag-view to ts

### PR DESCRIPTION
**Pull request by jeremy-rifkin**

### CODE REVIEW

File: static/panes/flags-view.js [hunk 1]
```diff
@@ -1,212 +0,0 @@
-// Copyright (c) 2021, Tom Ritter
```
Comment (matsjla) [inline static/panes/flags-view.js:1]: Not sure how this works with copyright, cc @mattgodbolt

File: static/panes/flags-view.ts [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+import { FlagsViewState } from './flags-view.interfaces';
+import { Container } from 'golden-layout';
+import { MonacoPaneState } from './pane.interfaces';
+import { Settings, SiteSettings } from '../settings';
+
+export class Flags extends MonacoPane<monaco.editor.IStandaloneCodeEditor, FlagsViewState> {
+    debouncedEmitChange: ((e: boolean) => void) & Cancelable = (() => {}) as any;
+    cursorSelectionThrottledFunction: ((e: any) => void) & Cancelable;
+    lastChangeEmitted: string;
+    constructor(hub: any, container: Container, state: FlagsViewState & MonacoPaneState) {
```
Comment (matsjla) [inline static/panes/flags-view.ts:39]: Can be typed as Hub now (#3470 is merged)

**Review by matsjla — APPROVED** [review]
LGTM - not sure about the attribution change though

File: static/panes/flags-view.js (hunk 1 above)
Comment (jeremy-rifkin) [reply static/panes/flags-view.js:1]: For all the ts conversions I've done I've left the copyright as-is if it's not "Compiler Explorer Authors." This dilemma is very reminiscent of the ship of Theseus :P Additive definitely seems the safe way to go, I can update.

**Review by mattgodbolt — APPROVED** [review]
LGTM!

**Comment by mattgodbolt** [comment]
Now live: thanks!

**Pull request merged**